test/e2e/framework/statefulset_utils.go

```diff
@@ -193,11 +194,30 @@ func getStatefulSetPodNameAtIndex(index int, ss *apps.StatefulSet) string {
 	return fmt.Sprintf("%v-%v", ss.Name, index)
 }
 
+// scaleFromStatefulSet returns a scale subresource for a statefulset.
+func scaleFromStatefulSet(ss *apps.StatefulSet, count int32) (*extensions.Scale, error) {
+	return &extensions.Scale{
+		// TODO: Create a variant of ObjectMeta type that only contains the fields below.
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ss.Name,
+			Namespace: ss.Namespace,
+		},
+		Spec: extensions.ScaleSpec{
+			Replicas: count,
+		},
+		Status: extensions.ScaleStatus{
+			Replicas: ss.Status.Replicas,
+			Selector: ss.Spec.Selector,
+		},
+	}, nil
+}
+
 // Scale scales ss to count replicas.
 func (s *StatefulSetTester) Scale(ss *apps.StatefulSet, count int32) error {
 	name := ss.Name
 	ns := ss.Namespace
-	s.update(ns, name, func(ss *apps.StatefulSet) { *(ss.Spec.Replicas) = count })
+	newScale, _ := scaleFromStatefulSet(ss, count)
+	s.update(ns, name, func(ss *apps.StatefulSet) { *(ss.Spec.Replicas) = newScale.Spec.Replicas })
```

Comment:
Sorry for not diving deeper enough before asking the question. To answer my own question, i should:
(1) enable `v1beta2` at `master.go`
(2) enable `Scale` type at `pkg/apis/apps/types.go` and `staging/src/k8s.io/api/apps/v1beta2/types.go` by putting `// +genclient=true` annotation
(3) run `hack/update-codegen.sh` to regenerate clientset

Autogen is required as we need to enable `v1beta2` to test the subresource. I am not sure this is the right approach, but will look at this direction. :)